### ISSUE TITLE
Playback doesn't resume after closing the player.

### ISSUE THREAD

**Issue by maaldheefee** [iwalton3/jellyfin-mpv-shim#74]
**Describe the bug**
Stopping the player (q or close) seems to save the progress bar on the movie or episode. But when played again, it doesn't resume from the last saved position. It starts from the position saved in the before opening Jellyfin MPV Desktop.

However, if the whole application is closed and started again, it resumes correctly.

**To Reproduce**
1. Open any media and play it.
2. Close the player using the close button or q key.
3. Play it again. It will start from the start or the position saved previously before starting step 1.

If after step 2, the whole thing (Jellyfin MPV Desktop) is closed and started again, and the same media is played, it starts from the correct position.

**Expected behavior**
Reopening the file should resume from the last position.

**Screenshots**
Not applicable

**Desktop (please complete the following information):**
 - OS: Windows 10 2004
 - Version 1.5.10

**Error Messages**
Please provide the console output by clicking the systray icon and selecting "Show Console".

**Comment by iwalton3** [comment]
> Another issue is that w key sets it as watched but doesn't close the player.

This key should mark the current video as watched and play the next one. (Maybe I didn't document it correctly somewhere?)

> However, if the whole application is closed and started again, it resumes correctly.

I have a suspicion that this is from the bundled web client and not the player backend. To confirm, could you try this with the Shim mode, casting from a web browser. Also try playing something on desktop mode, then search for something (to navigate away from the home page), and then go back to the home page and try to resume.

**Comment by maaldheefee** [comment]
> This key should mark the current video as watched and play the next one. (Maybe I didn't document it correctly somewhere?)

The w key is working for episodes. My mistake. I was using it on a movie, so I didn't see anything happening when I pressed it. It marked the movie as watched and continued playing.

> I have a suspicion that this is from the bundled web client and not the player backend.

When casting from the browser, it works perfectly, as expected.

Navigating away from the home page and going back also works perfectly.

So I'll stick to the Shim mode. (Is one method better than the other?)

Thanks for the help.

**Comment by iwalton3** [comment]
> Is one method better than the other?

The Shim mode was developed first and is lower resource overhead. I added the Desktop mode because it is easier to use. The playback engine is identical for both though.

The problem is likely that the desktop client's web app is outdated. (I maintain a patch set against the official web app for login and auto-cast, so it is kind of a pain to update.) I already need to mess with the web app for some other issues, so I'll probably update it when I look at those.

**Comment by maaldheefee** [comment]
Actually the issue is there when playing on the official web client too. So the issue is not on you side. Can you please let me know if this is happening for you too.

**Comment by iwalton3** [comment]
I can confirm that this is a problem in the current stable web client version. It seems to work intermittently, but I have seen it fail to start from the correct time.
